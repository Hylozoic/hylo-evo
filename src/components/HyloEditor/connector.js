import { connect } from 'react-redux'
import { findMentions, clearMentions, getMentionResults } from './store'

export function mapStateToProps (state, props) {
  return {
    mentionResults: getMentionResults(state, props)
  }
}

export const mapDispatchToProps = {
  findMentions,
  clearMentions
}

export default connect(mapStateToProps, mapDispatchToProps)
