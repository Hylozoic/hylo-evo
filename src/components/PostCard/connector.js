import { connect } from 'react-redux'
import { push } from 'react-router-redux'

export function mapStateToProps (state, props) {
  return {}
}

export const mapDispatchToProps = {
  navigate: push
}

export default connect(mapStateToProps, mapDispatchToProps)
