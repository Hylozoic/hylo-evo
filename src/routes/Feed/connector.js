import { connect } from 'react-redux'
import { SAMPLE_FEED_ITEMS } from './sampleData'

function mapStateToProps (state) {
  return {
    feedItems: SAMPLE_FEED_ITEMS
  }
}

function mapDispatchToProps (dispatch, props) {
  return {}
}

export default connect(mapStateToProps, mapDispatchToProps)
