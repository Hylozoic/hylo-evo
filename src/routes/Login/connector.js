import { connect } from 'react-redux'
import { checkLogin, login } from './actions'
export default connect(null, {checkLogin, login})
