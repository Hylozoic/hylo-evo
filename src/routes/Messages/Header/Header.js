import React from 'react'
import { isEmpty, filter, get, map } from 'lodash/fp'
import Icon from 'components/Icon'
import CloseMessages from '../CloseMessages'
import { others } from 'store/models/MessageThread'
import '../Messages.scss'

const MAX_CHARACTERS = 60

export default class Header extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      showAll: false,
      messageThreadId: null
    }
  }

  toggleShowAll = () => {
    const { showAll } = this.state
    this.setState({
      ...this.state,
      showAll: !showAll
    })
  }

  resetStateWithNewId = (messageThreadId) => {
    this.setState({
      showAll: false,
      messageThreadId
    })
  }

  componentWillReceiveProps = () => {
    const messageThreadId = get('id', this.props.messageThread)

    if (this.state.messageThreadId !== messageThreadId) {
      this.resetStateWithNewId(messageThreadId)
    }
  }

  getOthers ({ currentUser, messageThread }) {
    const participants = get('participants', messageThread) || []

    if (!currentUser) return participants

    const id = get('id', currentUser)

    return currentUser && map('name', filter(f => f.id !== id, participants))
  }

  render () {
    const { showAll } = this.state
    const { onCloseURL } = this.props
    const otherParticipants = this.getOthers(this.props)
    const maxShown = calculateMaxShown(showAll, otherParticipants, MAX_CHARACTERS)
    const { displayNames, andOthers } = generateDisplayNames(maxShown, otherParticipants)
    const showArrow = !!andOthers

    return <div styleName='header' id='thread-header'>
      <div styleName='header-text'>
        {displayNames}
        {andOthers && 'and' && <span styleName='toggle-link' onClick={this.toggleShowAll}>{andOthers}</span>}
      </div>
      {showArrow && !showAll && <Icon name='ArrowDown' styleName='arrow-down' onClick={this.toggleShowAll} />}
      {showAll && <Icon name='ArrowUp' styleName='arrow-up' onClick={this.toggleShowAll} />}
      <CloseMessages onCloseURL={onCloseURL} />
    </div>
  }
}

export function calculateMaxShown (showAll, otherParticipants, maxCharacters) {
  if (showAll) return otherParticipants.length
  if (!otherParticipants) return 0
  let count = 0
  for (let i = 0; i < otherParticipants.length; i++) {
    count += otherParticipants[i].length
    if (count > maxCharacters) {
      return i
    }
  }
  return otherParticipants.length
}

export function generateDisplayNames (maxShown, otherParticipants) {
  return isEmpty(otherParticipants)
    ? {displayNames: 'You'}
    : formatNames([...otherParticipants], maxShown)
}

export function formatNames (otherParticipants, maxShown) {
  let andOthers = null
  const length = otherParticipants.length
  const truncatedNames = (maxShown && maxShown < length)
    ? otherParticipants.slice(0, maxShown).concat([others(length - maxShown)])
    : otherParticipants
  if (maxShown && maxShown !== length) andOthers = truncatedNames.pop()

  if (andOthers) {
    return {displayNames: truncatedNames.join(', '), andOthers: ` ${andOthers}`}
  } else {
    return {displayNames: truncatedNames.join(', ')}
  }
}
