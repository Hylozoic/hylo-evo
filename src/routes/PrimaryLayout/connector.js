import { connect } from 'react-redux'
import { SAMPLE_COMMUNITY } from 'routes/Feed/sampleData'
import { fetchCurrentUser } from './actions'

function mapStateToProps ({currentUser}) {
  return {
    community: SAMPLE_COMMUNITY,
    currentUser
  }
}

export default connect(mapStateToProps, {fetchCurrentUser})
