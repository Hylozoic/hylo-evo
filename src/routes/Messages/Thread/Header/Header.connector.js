import { connect } from 'react-redux'
import { getOthers, getThreadId } from './Header.store'

export function mapStateToProps (state, props) {
  return {
    otherParticipants: getOthers(props),
    threadId: getThreadId(props)
  }
}

export default connect(mapStateToProps)
