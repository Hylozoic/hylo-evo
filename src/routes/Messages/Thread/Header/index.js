import Header from './Header'
import connector from './Header.connector'
export default connector(Header)
