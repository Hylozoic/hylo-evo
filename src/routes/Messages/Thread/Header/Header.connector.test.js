import { mapStateToProps } from './Header.connector'
describe('mapStateToProps', () => {
  it('returns the right keys', () => {
    const participants = [
      {name: 'one', id: 1},
      {name: 'two', id: 2},
      {name: 'three', id: 3}
    ]

    const thread = {
      id: 1,
      participants
    }
    const props = {
      currentUser: {id: 1},
      thread,
      threadId: thread.id
    }
    expect(mapStateToProps({}, props)).toMatchSnapshot()
  })
})
