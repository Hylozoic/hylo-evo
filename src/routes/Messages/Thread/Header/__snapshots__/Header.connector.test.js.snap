// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapStateToProps returns the right keys 1`] = `
Object {
  "otherParticipants": Array [
    "two",
    "three",
  ],
  "threadId": 1,
}
`;
